pandas/tests/scripts/test_validate_docstrings.py

```diff
@@ -0,0 +1,344 @@
+            Pattern to check for within each element.
+        case : bool, default True
+            Whether check should be done with case sensitivity.
+        na : object, default np.nan
+            Fill value for missing data.
+
+        Examples
+        --------
+        >>> s = pd.Series(['Antelope', 'Lion', 'Zebra', np.nan])
+        >>> s.str.contains(pat='a')
+        0    False
+        1    False
+        2     True
+        3      NaN
+        dtype: object
+
+        **Case sensitivity**
+
+        With `case_sensitive` set to `False` we can match `a` with both
+        `a` and `A`:
+
+        >>> s.str.contains(pat='a', case=False)
+        0     True
+        1    False
+        2     True
+        3      NaN
+        dtype: object
+
+        **Missing values**
+
+        We can fill missing values in the output using the `na` parameter:
+
+        >>> s.str.contains(pat='a', na=False)
+        0    False
+        1    False
+        2     True
+        3    False
+        dtype: bool
+        """
+        pass
+
+
+class BadDocStrings(object):
+
+    def func(self):
+
+        """Some function.
+
+        With several mistakes in the docstring.
+
+        It has a blank like after the signature `def func():`.
+
+        The text 'Some function' should go in the line after the
+        opening quotes of the docstring, not in the same line.
+
+        There is a blank line between the docstring and the first line
+        of code `foo = 1`.
+
+        The closing quotes should be in the next line, not in this one."""
+
+        foo = 1
+        bar = 2
+        return foo + bar
+
+    def astype(self, dtype):
+        """
+        Casts Series type.
+
+        Verb in third-person of the present simple, should be infinitive.
+        """
+        pass
+
+    def astype1(self, dtype):
+        """
+        Method to cast Series type.
+
+        Does not start with verb.
+        """
+        pass
+
+    def astype2(self, dtype):
+        """
+        Cast Series type
+
+        Missing dot at the end.
+        """
+        pass
+
+    def astype3(self, dtype):
+        """
+        Cast Series type from its current type to the new type defined in
+        the parameter dtype.
+
+        Summary is too verbose and doesn't fit in a single line.
+        """
+        pass
+
+    def plot(self, kind, **kwargs):
+        """
+        Generate a plot.
+
+        Render the data in the Series as a matplotlib plot of the
+        specified kind.
+
+        Note the blank line between the parameters title and the first
+        parameter. Also, note that after the name of the parameter `kind`
+        and before the colon, a space is missing.
+
+        Also, note that the parameter descriptions do not start with a
+        capital letter, and do not finish with a dot.
+
+        Finally, the `**kwargs` parameter is missing.
+
+        Parameters
+        ----------
+
+        kind: str
+            kind of matplotlib plot
```

Comment:
I would try to split up some of those "bad" methods so each function is focused on 1 bad element, to be sure we are testing that.

So for example, the plot one above could be split up in:

```
    def test_missing_kwargs(self, kind, **kwargs):
        """
        Kwargs argument is missing in parameters section.


        Parameters
        ----------
        kind : str
            Kind of matplotlib plot.
        """

    def test_parameter_colon_spacing(self, kind):
        """
        Space around colon is missing.

        Parameters
        ----------
        kind: str
            Kind of matplotlib plot.
        """

    def test_parameter_description_period(self, kind):
        """
        Parameter description does not end in a dot.

        Parameters
        ----------
        kind : str
            Kind of matplotlib plot
        """

    def test_parameter_description_capital(self, kind):
        """
        The parameter descriptions do not start with a capital.

        Parameters
        ----------
        kind : str
            kind of matplotlib plot.
        """

    def test_parameter_blank_line(self, kind):
        """
        Blank line between the parameters title and the first parameter. 
    
        Parameters
        ----------
        
        kind : str
            Kind of matplotlib plot.
        """
```

Reply:
Yes this is a great idea and what I envision for this script long term (would make it clearer the purpose of the tests). 

For the initial change here I was just copying from the sprint docs, but I'll look how to integrate the above and any others that come to mind